src/mujoco_system_interface.cpp

```
@@ -973,29 +1088,161 @@ void MujocoSystemInterface::register_joints(const hardware_interface::HardwareIn
+            const auto gains = last_joint_state.pos_pid->get_gains();
+
+            RCLCPP_INFO(rclcpp::get_logger("MujocoSystemInterface"),
+                        "Position control PID gains for joint %s : P=%.4f, I=%.4f, D=%.4f, Imax=%.4f, Imin=%.4f, "
+                        "Umin=%.4f, Umax=%.4f, antiwindup_strategy=%s",
+                        joint.name.c_str(), gains.p_gain_, gains.i_gain_, gains.d_gain_, gains.antiwindup_strat_.i_max,
+                        gains.antiwindup_strat_.i_min, gains.u_min_, gains.u_max_,
+                        gains.antiwindup_strat_.to_string().c_str());
+          }
+          else
+          {
+            RCLCPP_ERROR(rclcpp::get_logger("MujocoSystemInterface"),
+                         "Position command interface for the joint : %s is not supported with velocity or motor "
+                         "actuator without defining the PIDs",
+                         joint.name.c_str());
+            continue;
+          }
+        }
+        else if (last_joint_state.actuator_type == ActuatorType::POSITION)
+        {
+          last_joint_state.is_position_control_enabled = true;
+          last_joint_state.position_command =
+              should_override_start_position ? mj_data_->ctrl[joint_state.mj_actuator_id] : last_joint_state.position;
+        }
       }
       else if (command_if.name.find(hardware_interface::HW_IF_VELOCITY) != std::string::npos)
       {
-        last_joint_state.is_velocity_control_enabled = true;
-        last_joint_state.velocity_command =
-            should_override_start_position ? mj_data_->ctrl[joint_state.mj_actuator_id] : last_joint_state.velocity;
+        if (last_joint_state.actuator_type == ActuatorType::POSITION)
+        {
+          RCLCPP_ERROR(rclcpp::get_logger("MujocoSystemInterface"),
+                       "Velocity command interface for the joint : %s is not supported with position actuator",
+                       joint.name.c_str());
+          continue;
```

These should all probably be failure cases rather than ignored.

---

Better to add some error, so that it is easy for the end user to understand the issue